Multiple fixes on Always Incognito Mode

uazo opened the pull request:
1. added specific menu accessible from privacy settings with options:
- Enable always incognito (default false)
- Enable history support for incognito mode (default false)
- Allow the saving of the site settings for incognito mode (default false)

I've also added a link for online help to https://github.com/bromite/bromite/wiki/AlwaysIncognito

2. merged [patch](https://github.com/uazo/bromite/blob/master-v101/build/patches/contrib/Add-history-support-in-incognito-mode-addon.patch) which disable the support for automatic saving of pages for offline mode in always incognito mode, currently active by mistake with the saving of the history

3. added new `kOfflinePagesAutoSaveFeature` flag (`offline-pages-auto-save` from about_flags) which allows to disable the option also for normal mode, useful for speeding up the return from Android Recents with heavy pages (or with heavy userscripts, see https://github.com/bromite/bromite/issues/792#issuecomment-1062648912 @xarantolus)

4. removed test code, currently useless

5. fix incorrect prefs registration position (caused by rebase v101)

6. Removed useless `IsAlwaysIncognito()` in `HostContentSettingsMap`

Fix for #1991

csagan5 commented:
> 2\. merged [patch](https://github.com/uazo/bromite/blob/master-v101/build/patches/contrib/Add-history-support-in-incognito-mode-addon.patch)

Link is broken.

> * removed test code, currently useless

Sad to see them go, I think the idea was right.

> 3\. added new `kOfflinePagesAutoSaveFeature` flag (`offline-pages-auto-save` from about_flags) which allows to disable the option also for normal mode

I am thinking that perhaps this should be off by default, the user should ask to save, not automatically.

Can you also move "Open external links in incognito" in this new page? I will merge the patch for now so please make sure to get the updated version before that.

uazo commented:
> > 2. merged [patch](https://github.com/uazo/bromite/blob/master-v101/build/patches/contrib/Add-history-support-in-incognito-mode-addon.patch)
> 
> Link is broken.

updated

> > * removed test code, currently useless
> 
> Sad to see them go, I think the idea was right.

yes, but i think it's useless to keep it aligned in rebases if i don't use it.
i like the method used by @arthuredelstein more, which i will try sooner or later, maybe upgrading webdriver if necessary

> > 3. added new `kOfflinePagesAutoSaveFeature` flag (`offline-pages-auto-save` from about_flags) which allows to disable the option also for normal mode
> 
> I am thinking that perhaps this should be off by default, the user should ask to save, not automatically.

from my point of view there is no problem, I always use the always incognito mode and I have never found that need.

> Can you also move "Open external links in incognito" in this new page? I will merge the patch for now so please make sure to get the updated version before that.

Sure.
